utils/gatekeeper.py

```diff
@@ -0,0 +1,33 @@
+from flask import current_app, request
+from functools import wraps
+from typing import Set
+
+
+def allowed_params(params: Set[str]):
+    def decorator(endpoint):
+        @wraps(endpoint)
+        def verify_fields(*args, **kwargs):
+            if not (request.json.keys() <= params):
+                return invalid_field_error(params)
+            return endpoint(*args, **kwargs)
+
+        return verify_fields
+
+    return decorator
+
+
+def invalid_field_error(field_set):
+    if current_app.env == "production":
+        return {"message": "Invalid request field"}, 400
```

Comment:
Another option here would be to return the endpoint with an argument that consisted of only the subset of allowed params, essentially filtering out out any unwanted params. And then we can always use that argument in the action instead of request.json - I'm not sure if that's possible though.